src/service/resources-impl.js

```diff
@@ -912,6 +925,29 @@ export class Resources {
     dev().fine(TAG_, 'element upgraded:', resource.debugid);
   }
 
+  /**
+   * Schedule resources for preconnecting.
+   * @param {!Element} element
+   */
+  requestPreconnect(element) {
```

Comment:
I feel like we can do a better optimization by applying this at the `Timer` level and not on the `preconnect` method specifically.